jaxrs/src/main/java/org/jboss/as/jaxrs/DeploymentRestResourcesDefintion.java

```diff
@@ -184,71 +187,88 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
-                                                Class<?> resClsInModel;
-                                                try {
-                                                    resClsInModel = Class.forName(clsName, false, resClass.getClassLoader());
-                                                } catch (ClassNotFoundException e) {
-                                                    throw new OperationFailedException(e);
-                                                }
-                                                if (resClass.equals(resClsInModel)
-                                                        || (resClass.isInterface() && resClass.isAssignableFrom(resClsInModel) && !clsInResourceInvokers(registry.getBounded(), resClsInModel))) {
-                                                    JaxrsResourceMethodDescription resMethodDesc = resMethodDescription(methodInvoker, contextPath, mapping, servletMappings, clsName);
-                                                    resMethodInvokers.add(resMethodDesc);
-                                                }
-                                            } else if (ResourceLocatorInvoker.class.isAssignableFrom(resourceInvoker.getClass())) {
-                                                ResourceLocatorInvoker locatorInvoker = (ResourceLocatorInvoker) resourceInvoker;
-                                                if (clsName.equals(locatorInvoker.getMethod().getDeclaringClass().getCanonicalName())) {
-                                                    ResourceClass resClass = ResourceBuilder.locatorFromAnnotations(locatorInvoker.getMethod().getReturnType());
-                                                    JaxrsResourceLocatorDescription resLocatorDesc = resLocatorDescription(resClass, contextPath, mapping, servletMappings, new ArrayList<Class<?>>());
-                                                    resLocatorInvokers.add(resLocatorDesc);
+                                    for (HttpServletDispatcher resteasyServlet: resteasyServlets) {
+                                        final Collection<String> servletMappings = resteasyServlet.getServletConfig().getServletContext()
+                                                .getServletRegistration(resteasyServlet.getServletConfig().getServletName()).getMappings();
+                                        final ResourceMethodRegistry registry = (ResourceMethodRegistry) resteasyServlet.getDispatcher().getRegistry();
+                                                List<JaxrsResourceMethodDescription> resMethodInvokers = new ArrayList<>();
+                                                List<JaxrsResourceLocatorDescription> resLocatorInvokers = new ArrayList<>();
+                                                for (Map.Entry<String, List<ResourceInvoker>> resource : registry
+                                                        .getBounded().entrySet()) {
```

Comment:
This would have been quite a bit easier to review if these line break changes weren't there.

A secret magic trick for looking at these is to add ?w=1 to the github URL:

https://github.com/wildfly/wildfly/pull/12138/files?w=1

If you do that github will display changes due to whitespace very differently; what seems like really large changes because blocks of code got indented to another level, like there are here, disappear.

Once you do that and look at the change in this file, a whole lot of the remaining changes are just due to these different line breaks. Eliminate those and this complex scary change suddenly is quite simple. You're looking up the deploymentModule so you can load from it, and you're processing a List<HttpServletDispatcher> resteasyServlets instead of Servlet resteasyServlet. And you throw ISE instead of OFE if a CNFE is caught.

I understand why the line breaks were added; indenting another level pushes a lot of code off the screen.

@ronsigal FYI, in case that makes it easier to see what this is doing.